uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/leave-request-popup/details-tab/toil.html

```diff
@@ -39,24 +39,34 @@
       <div class="col-xs-12">
         <label class="control-label required-mark">TOIL to be Accrued</label>
       </div>
-      <div class="col-xs-12">
-        <div
-          class="crm_custom-select crm_custom-select--full"
-          ng-if="detailsTab.selectedAbsenceType.calculation_unit_name === 'days'">
-          <select
-            class="no-select2"
-            name="toilAmount"
-            ng-disabled="detailsTab.isMode('view')"
-            ng-options="amount.value as amount.label for amount in detailsTab.toilAmounts"
-            ng-model="detailsTab.request.toil_to_accrue"
-            ng-change="detailsTab.performBalanceChangeCalculation()">
-            <option value="">- select -</option>
-          </select>
-          <span class="crm_custom-select__arrow"></span>
+      <div
+        ng-if="detailsTab.selectedAbsenceType.calculation_unit_name === 'days'"
+        ng-init="showTOILAccrualsOptionEditorIcon = $root.section === 'admin-dashboard' || $root.section === 'absence-tab'">
+        <div ng-class="{ 'col-xs-11': showTOILAccrualsOptionEditorIcon, 'col-xs-12': !showTOILAccrualsOptionEditorIcon }">
+          <div
+            class="crm_custom-select crm_custom-select--full"
+            ng-if="detailsTab.selectedAbsenceType.calculation_unit_name === 'days'">
+            <select
+              class="no-select2"
+              name="toilAmount"
+              ng-disabled="detailsTab.isMode('view')"
+              ng-options="amount.value as amount.label for amount in detailsTab.toilAmounts"
+              ng-model="detailsTab.request.toil_to_accrue"
+              ng-change="detailsTab.performBalanceChangeCalculation()">
+              <option value="">- select -</option>
+            </select>
+            <span class="crm_custom-select__arrow"></span>
+          </div>
         </div>
+        <div ng-if="showTOILAccrualsOptionEditorIcon" class="col-xs-1 control-label">
+          <a class="pointer">
+            <i class="crm-i fa-wrench" ng-click="detailsTab.openToilInDaysAccrualOptionsEditor()"></i>
+          </a>
+        </div>
+      </div>
+      <div class="col-xs-12" ng-if="detailsTab.selectedAbsenceType.calculation_unit_name === 'hours'">
```

Comment:
seems you have this condition multiple times in the view, I think you should put it in the controller, something like
```html
<div ng-if="isCalculationIn('hours')">
```

Reply:
Done.